packages/relay/src/lib/precheck.ts

```diff
@@ -61,7 +62,10 @@ export class Precheck {
    * @param {Transaction} tx - The transaction.
    */
   value(tx: Transaction): void {
-    if (tx.data === EthImpl.emptyHex && tx.value < constants.TINYBAR_TO_WEIBAR_COEF) {
+    if (
+      tx.data === EthImpl.emptyHex &&
+      ((tx.value > 0 && tx.value < constants.TINYBAR_TO_WEIBAR_COEF) || tx.value < 0)
+    ) {
```

Comment:
Don't we also want to perform this pre-check on non-zero data transaction? 

What if the solidity method is marked as payable and we pass a negative value or a value between 0 and `TINYBAR_TO_WEIBAR_COEF`? Don't we want to confirm that the passed value is correct also in this case?